File: 0014-longest-common-prefix/0014-longest-common-prefix.py

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        prefix = strs[0]  # Start with the first word as the prefix
        
        for word in strs[1:]:  # Compare with remaining words
            i = 0
            while i < len(prefix) and i < len(word) and prefix[i] == word[i]:
                i += 1
            prefix = prefix[:i]  # Trim the prefix to the matching part
            
            if not prefix:  # If prefix becomes empty, return ""
                return ""
        
        return prefix

File: 0014-longest-common-prefix/0014-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    prefix = ""
    length = strs.map { |s| s.length }.min
    0.upto(length - 1) do |i|
        char = strs[0][i]
        strs.each { |s| return prefix if s[i] != char }
        prefix += char
    end
    prefix
end